vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,292 @@
+---
+title: "Projecting infectious disease incidence using early outbreak data"
+author: "James Azam, Sebastian Funk"
+output:
+  bookdown::html_vignette2:
+    fig_caption: yes
+    code_folding: show
+pkgdown:
+  as_is: true
+bibliography: references.bib
+link-citations: true
+vignette: >
+  %\VignetteIndexEntry{Projecting infectious disease incidence using early outbreak data}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, 
+                      message = FALSE, 
+                      warning = FALSE, 
+                      collapse = TRUE,
+                      comment = "#>",
+                      dpi = 300
+                      )
+
+```
+
+## Overview
+
+Branching processes can be used to project infectious disease trends provided 
+we can characterize the distribution of times between 
+successive cases (serial interval), and the distribution of secondary cases 
+produced by a single individual (offspring distribution). Such simulations can 
+be achieved in `bpmodels` with the `chain_sim()` function. @Pearson2020, and 
+@abbott2020 illustrate its application to COVID-19. 
+
+The purpose of this vignette is to use early data on COVID-19 in South Africa 
+[@marivate2020] to illustrate how `bpmodels` can be used to forecast an 
+outbreak. 
+
+
+Let's load the required packages
+
+```{r packages, include=TRUE}
+library("bpmodels")
+library("dplyr")
+library("ggplot2")
+library("lubridate")
+```
+
+## Data
+
+We will get and clean the first $15$ days of the COVID-19 
+outbreak in South Africa to seed the simulation for this example.
+
+```{r data, message=FALSE}
+data_url <- "https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_timeline_confirmed.csv" # nolint: line_length_linter. 
+
+#Read the data in using the url
+covid19_sa <- read.csv(data_url)
+
+# Subset the first 15 days and count the number of cases per date
+covid19_sa <- covid19_sa %>% 
+  select(date) %>% 
+  mutate(date = lubridate::dmy(date)) %>%
+  filter(date <= min(date) + lubridate::days(15)) %>%   
+  group_by(date) %>% 
+  summarise(cases = n()) %>%   
+  ungroup()
+```
+
+## Inputs  
+Using the data above, we will set up a vector of start times for each case.
+
+```{r linelist_gen, message=FALSE}
+days_since_index <- as.integer(covid19_sa$date - min(covid19_sa$date))
+
+start_times <- unlist(mapply(
+  function(x, y) rep(x, times = ifelse(y == 0, 1, y)),
+  days_since_index,
+  covid19_sa$cases
+))
+                       
+```
+
+
+Additionally, `chain_sim()` requires the end time for the simulations and the 
+maximum size of each chain. Since each result of `chain_sim()` is stochastic,
+it is also best to run it many times. 
+
+We will specify these as follows:
+
+```{r input_prep2, message=FALSE}
+#' Date to end simulation (14 day projection in this case)
+
```

Comment:
```suggestion
#' Date to end simulation (14 day projection in this case)
```

remove new line - comment belongs to next line

Reply:
Done.